core/domain/skill_services.py

```diff
@@ -586,6 +657,115 @@ def save_skill_summary(skill_summary):
+        creator_id=skill_rights.creator_id,
+        skill_is_private=skill_rights.skill_is_private
+    ).commit(committer_id, 'Created new skill rights', commit_cmds)
+
+
+def get_skill_rights_from_model(skill_rights_model):
+    """Constructs a SkillRights object from the given skill rights model.
+
+    Args:
+        skill rights model: SkillRightsModel. Skill rights from the datastore.
+
+    Returns:
+        SkillRights. The rights object created from the model.
+    """
+
+    return skill_domain.SkillRights(
+        skill_rights_model.id,
+        skill_rights_model.skill_is_private,
+        skill_rights_model.creator_id
+    )
+
+
+def get_skill_rights(skill_id, strict=True):
+    """Retrieves the rights object for the given skill.
+
+    Args:
+        skill_id: str. ID of the skill.
+        strict: bool. Whether to fail noisily if no skill with the given id
+            exists in the datastore.
+
+    Returns:
+        SkillRights. The rights object associated with the given skill.
+
+    Raises:
+        EntityNotFoundError. The skill with ID skill id was not found
+            in the datastore.
+    """
+
+    model = skill_models.SkillRightsModel.get(skill_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_skill_rights_from_model(model)
+
+
+def get_unpublished_skills_by_creator(user_id):
+    skill_rights_models = (
+        skill_models.SkillRightsModel.get_unpublished_skills_by_creator_id(
+            user_id))
+    return [get_skill_rights_from_model(skill_rights_model)
+            for skill_rights_model in skill_rights_models]
+
+
+def check_can_edit_skill(user, skill_rights):
+    """Checks whether the user can edit the given skill.
+
+    Args:
+        user: UserActionsInfo. Object having user id, role and actions for
+            given user.
+        skill_rights: SkillRights or None. Rights object for the given skill.
+
+    Returns:
+        bool. Whether the given user can edit the given skill.
+    """
+    if skill_rights is None:
+        return False
+    if role_services.ACTION_EDIT_PUBLIC_SKILLS not in user.actions:
+        return False
+    if role_services.ACTION_EDIT_PUBLIC_SKILLS in user.actions:
+        if not skill_rights.is_private():
+            return True
+        if skill_rights.is_private() and skill_rights.is_creator(user.user_id):
+            return True
+    return False
+
+
+def check_can_publish_skill(user, skill_rights):
+    """Checks whether the user can publish the given skill.
+
+    Args:
+        user: UserActionsInfo. Object having user id, role and actions for
+            given user.
+        skill_rights: SkillRights or None. Rights object for the given skill.
+
+    Returns:
+        bool. Whether the given user can publish the given skill.
+    """
+
+    if skill_rights is None:
+        return False
+    if role_services.ACTION_PUBLISH_OWNED_SKILL not in user.actions:
+        return False
+    if skill_rights.is_creator(user.user_id):
```

Comment:
Do you also need to check that the skill is currently private?

Reply:
Not sure, but I feel like that doesn't have much to do with ACL rather the state of the skill